homeassistant/components/godice/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Sensors for GoDice device."""
+
+import logging
+
+import godice
+
+from homeassistant import const
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_DEVICE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ROLLED_NUMBER_SENSOR_DESCR = SensorEntityDescription(
+    key="rolled_number", state_class=SensorStateClass.MEASUREMENT
+)
+COLOR_SENSOR_DESCR = SensorEntityDescription(
+    key="color",
+    device_class=SensorDeviceClass.ENUM,
+    options=[color.name for color in godice.Color],
+)
+BATTERY_SENSOR_DESCR = SensorEntityDescription(
+    key="battery_level",
+    device_class=SensorDeviceClass.BATTERY,
+    native_unit_of_measurement=const.PERCENTAGE,
+    state_class=SensorStateClass.MEASUREMENT,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Setups Dice sensors."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    device = data[DATA_DEVICE]
+    device_name = config_entry.data[const.CONF_NAME]
+    device_info = DeviceInfo(
+        identifiers={(DOMAIN, device_name)},
+        name=device_name,
+        manufacturer="Particula",
+        model="GoDice",
+    )
```

Comment:
You can use the bluetooth address as the DeviceInfo identifier, which you can get from the BLEDevice. Also, I would put all strings like manufacturer and model in const.py so people know where to look for these constants.
```suggestion
    from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH
    device_info = DeviceInfo(
        identifiers={(CONNECTION_BLUETOOTH, bluetooth_address)},
        name=device_name,
        manufacturer=MANUFACTURER,
        model=MODEL,
    )
```

Reply:
creating `DeviceInfo` once and passing the same object for all sensors better demonstrates the fact that all sensors belong to the same device, in contrast of creating `DeviceInfo` per each Sensor